code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -288,73 +287,92 @@
+				if(beaker && dispensable_reagents.Find(r_id)) // but since we verify we have the reagent, it'll be fine
+					var/datum/reagents/R = beaker.reagents
+					var/free = R.maximum_volume - R.total_volume
+					var/rounded = is_resolution(chemicals_to_dispense[key]) ? chemicals_to_dispense[key] : round(chemicals_to_dispense[key], macroresolution)
+					var/actual = min(rounded, (cell.charge * powerefficiency)*10, free)
+					if(actual)
+						if(!cell.use(abs(actual) / powerefficiency))
+							say("Not enough energy to complete operation!")
+							return
+						if(actual > 0)
+							R.add_reagent(r_id, actual)
+						if(actual < 0)
+							R.remove_reagent(r_id, abs(actual))
+						work_animation()
 		if("clear_recipes")
 			if(!is_operational)
 				return
 			var/yesno = tgui_alert(usr, "Clear all recipes?",, list("Yes","No"))
 			if(yesno == "Yes")
 				saved_recipes = list()
 			. = TRUE
-		if("record_recipe")
+		if("record_recipe") //I'm not renaming this lol
 			if(!is_operational)
 				return
-			recording_recipe = list()
-			. = TRUE
-		if("save_recording")
-			if(!is_operational)
-				return
-			var/name = stripped_input(usr,"Name","What do you want to name this recipe?", "Recipe", MAX_NAME_LEN)
+			var/name = stripped_input(usr,"Name","What do you want to name this recipe?(This is broken)", "Recipe", MAX_NAME_LEN)
+			var/recipe = stripped_input(usr,"Recipe","Insert recipe with chem IDs")
 			if(!usr.canUseTopic(src, !issilicon(usr)))
 				return
-			if(saved_recipes[name] && tgui_alert(usr, "\"[name]\" already exists, do you want to overwrite it?",, list("Yes", "No")) == "No")
-				return
-			if(name && recording_recipe)
-				for(var/reagent in recording_recipe)
-					var/reagent_id = GLOB.name2reagent[translate_legacy_chem_id(reagent)]
-					if(!dispensable_reagents.Find(reagent_id))
-						visible_message(span_warning("[src] buzzes."), span_hear("You hear a faint buzz."))
-						to_chat(usr, "<span class ='danger'>[src] cannot find <b>[reagent]</b>!</span>")
-						playsound(src, 'sound/machines/buzz-two.ogg', 50, TRUE)
+			if(name && recipe)
+				var/list/first_process = splittext(recipe, ";")
+				if(!LAZYLEN(first_process))
+					return
+				var/resmismatch = FALSE
+				for(var/reagents in first_process)
+					var/list/reagent = splittext(reagents, "=")
+					var/amt = text2num(reagent[2]) || 0
+					var/reagent_id = GLOB.name2reagent[translate_legacy_chem_id(reagent[1])]
+					if(dispensable_reagents.Find(reagent_id) || (reagent_id && amt < 0))
+						if (!resmismatch && !check_macro_part(reagents))
+							resmismatch = TRUE
+						continue
+					else
+						var/chemid = reagent[1]
+						visible_message("<span class='warning'>[src] buzzes.</span>", "<span class='italics'>You hear a faint buzz.</span>")
+						to_chat(usr, "<span class='danger'>[src] cannot find Chemical ID: <b>[chemid]</b>!</span>")
+						playsound(src, 'sound/machines/buzz-two.ogg', 50, 1)
 						return
-				saved_recipes[name] = recording_recipe
```

Comment:
`saved_recipes[name] = recording_recipe` to `saved_recipes += list(list("recipe_name" = name, "contents" = recipe))`, This  is probably what's breaking it. Instead of adding a recipe value with the key of the recipe's name, you're adding a list that contains the list that contains the recipe name and the recipe iteslf as values. This is triply nested lists. `saved_recipes[name] = recipe` should ail your issues.